components/Discussion/enhancers.js

```diff
@@ -55,6 +55,74 @@ query discussionMe($discussionId: ID!) {
 `
 export const withMe = graphql(meQuery)
 
+// This function extends the props with the 'DisplayUser' that will be used
+// when the current user submits a new comment in the given discussion
+// (specified by its discussionId in the ownProps). The shape of the 'DisplayUser'
+// is determined based on the discussion rules, and the users' preferences.
+//
+// The prop is called 'discussionDisplayAuthor'.
+export const withDiscussionDisplayAuthor = graphql(gql`
+query discussionDisplayAuthor($discussionId: ID!) {
+  me {
+    id
+    name
+    publicUser {
+      testimonial {
+        image(size: SHARE)
+      }
+    }
+  }
+  discussion(id: $discussionId) {
+    id
+    rules {
+      anonymity
+    }
+    userPreference {
+      anonymity
+      credential {
+        description
+        verified
+      }
+    }
+  }
+}
+`, {
+  props: ({ownProps: {t}, data: {me, discussion}}) => {
+    if (!me || !discussion) {
+      return {}
+    }
+
+    const {rules, userPreference} = discussion
+
+    const anonymous = (() => {
+      switch (rules.anonymity) {
+        case 'ALLOWED': return userPreference ? userPreference.anonymity : false
+        case 'ENFORCED': return true
+        case 'FORBIDDEN': return false
+        default: return false
+      }
+    })()
+
+    if (anonymous) {
+      return {
+        discussionDisplayAuthor: {
+          name: t('discussion/displayUser/anonymous'),
```

Comment:
Yes, I'd rely on the server response here, so you'd just have to update
https://docs.google.com/spreadsheets/d/1NE1igcbxtTrVd5P72Ubqmgv4yKYh0LRGa0EuYSEQKxE/edit#gid=0
and use that message here.

Reply:
Ideally the whole 'DisplayUser' would be constructed by the server and sent to the client. There is a fair bit of logic here in the client that I'm sure is also implemented in the server already.

Once you extend the schema so that the client can fetch the display user, this enhancer is all you'll need to change.